packages/@lwc/engine/src/framework/decorators/wire.ts

```diff
@@ -35,11 +35,16 @@ export function internalWireFieldDecorator(key: string): PropertyDescriptor {
         set(this: ComponentInterface, value: any) {
             const vm = getAssociatedVM(this);
             /**
-             * intentionally ignoring the reactivity here since this is just
+             * Reactivity for wired fields is provided in wiring.
+             * We intentionally add reactivity here since this is just
              * letting the author to do the wrong thing, but it will keep our
              * system to be backward compatible.
              */
-            vm.cmpFields[key] = value;
+            if (value !== vm.cmpFields[key]) {
```

Comment:
Since we will be doing this, what about if when defining a [fieldDataCallback](https://github.com/salesforce/lwc/blob/master/packages/@lwc/engine/src/framework/wiring.ts#L19-L30), we rely on this wired field property descriptor?